TSDB head too far in future improvements (#11961)
grafana/mimir: docs/sources/mimir/manage/mimir-runbooks/_index.md

- The metric exported by the ingester computes the maximum timestamp from all TSDBs open in the ingester.
- The alert checks the metric and fires if the maximum timestamp is more than 1h in the future.
- This alert does not respect the per-user creation grace period.

How to **investigate**

> - Find the affected ingester pod, connect to its http API e.g. using `kubectl port-forward <pod> 8080:80`
> - Find the tenant with a bad sample on the ingesters tenants list (obtained via the `/ingester/tenants` endpoint), where a warning "TSDB Head max timestamp too far in the future" is displayed.
> - If there are no warnings, the sample is within the tenant's grace interval (false positive).
> - Flush the tenant's data to blocks storage (via `/ingester/flush?wait=true&tenant=foo` endpoint).

Reviewer: I think the following instructions are clearer:
```suggestion
- Find an affected ingester pod, connect to its http API e.g. using `kubectl port-forward <pod> 8080:80`
- Find the tenant with a bad sample on the ingester's tenants list (obtained via the `/ingester/tenants` endpoint), where a warning "TSDB Head max timestamp too far in the future" should be displayed.
- If there are no warnings, the sample is likely within the tenant's grace interval (false positive).

If it's not a false positive:
- Flush the tenant's data to blocks storage (via `/ingester/flush?wait=true&tenant=foo` endpoint).
- Remove the tenant's directory on disk and the restart ingester.
```

- The metric exported by the ingester computes the maximum timestamp from all TSDBs open in the ingester.
- The alert checks the metric and fires if the maximum timestamp is more than 1h in the future.

> - This alert does not respect the per-user creation grace period.

Reviewer 2: ```suggestion
- This alert doesn't respect the per-user creation grace period.
```

How to **investigate**

> - Find an affected ingester pod, connect to its http API e.g. using `kubectl port-forward <pod> 8080:80`

Reviewer 2: ```suggestion
- Find an affected ingester pod and connect to its http API. For example, using `kubectl port-forward <pod> 8080:80`
```

> - Find the tenant with a bad sample on the ingester's tenants list (obtained via the `/ingester/tenants` endpoint), where a warning "TSDB Head max timestamp too far in the future" should be displayed.

Reviewer 2: ```suggestion
- Find the tenant with a bad sample on the ingester's tenants list, which you can get through the `/ingester/tenants` endpoint. The sample displays the warning "TSDB Head max timestamp too far in the future".
```

> - If there are no warnings, the sample is likely within the tenant's grace interval (false positive).

Reviewer 2: ```suggestion
- If there are no warnings, the sample is likely within the tenant's grace interval, and the alert is a false positive.
```

If it's not a false positive:

> - Flush the tenant's data to blocks storage (via `/ingester/flush?wait=true&tenant=foo` endpoint).

Reviewer 2: ```suggestion
- Flush the tenant's data to blocks storage through the `/ingester/flush?wait=true&tenant=foo` endpoint.
```
